src/components/userGroups/K8sObjectPermissions/K8sPermissionModal.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useState } from 'react'
+import { ButtonWithLoader, VisibleModal } from '../../common'
+import { ActionTypes, OptionType } from '../userGroups.types'
+import { ReactComponent as Close } from '../../../assets/icons/ic-close.svg'
+import { ReactComponent as AddIcon } from '../../../assets/icons/ic-add.svg'
+import K8sListItemCard from './K8sListItemCard'
+
+const getEmptyPermissionObject = (idx = 0, k8sPermission = null) => {
+    return {
+        key: idx,
+        cluster: k8sPermission?.cluster,
+        namespace: k8sPermission?.namespace,
+        group: k8sPermission?.group,
+        kind: k8sPermission?.kind,
+        resource: k8sPermission?.resource,
+        action: k8sPermission?.action || { value: ActionTypes.VIEW, label: ActionTypes.VIEW },
+    }
+}
+
+export default function K8sPermissionModal({ selectedPermissionAction, k8sPermission, setK8sPermission, close }) {
+    const [k8PermissionList, setPermissionList] = useState([getEmptyPermissionObject(0, k8sPermission)])
+    const [namespaceMapping, setNamespaceMapping] = useState<Record<number, OptionType[]>>()
+    const [apiGroupMapping, setApiGroupMapping] = useState<Record<number, OptionType[]>>()
+    const [kindMapping, setKindMapping] = useState<Record<number, OptionType[]>>()
+    const [objectMapping, setObjectMapping] = useState<Record<number, OptionType[]>>()
+    const [isDataFilled, setIsDataFilled] = useState<boolean>(true)
+
+    useEffect(() => {
+        const disbale = k8PermissionList
+            ? k8PermissionList.find((item) => item.resource === null || item.resource?.length === 0)
+            : true
+        setIsDataFilled(!!disbale)
```

Comment:
Please change this logic to be used without use effect as discussed.

Reply:
done